### ISSUE TITLE
Add forward_by_train and forward_by_eval convenience methods to nn.Module

### ISSUE THREAD

**Issue by msdsm** [pytorch/pytorch#170851]
### Description
I propose adding two convenience methods to `nn.Module`: `forward_by_train()` and `forward_by_eval()`. These methods explicitly set the module's training mode before executing the forward pass, reducing boilerplate and preventing common errors where users forget to toggle `train()`/`eval()` modes.

### Motivation
It is a common pitfall in PyTorch to forget to call `.eval()` before inference or `.train()` before training. This leads to subtle but significant bugs, especially with layers like `BatchNorm` (running statistics) and `Dropout` (randomly zeroing elements). 

Currently, the standard workflow requires two separate steps:
1. Set the mode (`model.train()` or `model.eval()`)
2. Execute the forward pass (`model(input)`)

While seasoned users are familiar with this, it remains a frequent source of "silent bugs" for many. Providing an atomic method to "set mode and forward" makes the intent explicit and the code more robust and readable.

### Proposed API
Add the following methods to `torch.nn.Module`:

```python
def forward_by_train(self, *args: Any, **kwargs: Any) -> Any:
    self.train()
    return self(*args, **kwargs)

def forward_by_eval(self, *args: Any, **kwargs: Any) -> Any:
    self.eval()
    return self(*args, **kwargs)
```
